### PR TITLE
Theme/#414 a11y dropdowns

### LINKED ISSUE #414: a11y dropdown menus
Example: https://codeable.io/wordpress-accessibility-creating-accessible-dropdown-menus/

### PR DESCRIPTION
**Pull request by mharis**
Closes #414 

### DESCRIPTION ###
Adding accessibility to the menus on `wd_s` including dropdowns.

- Added labels to menu areas.
- Off-canvas menu is now focused when it's visible. This will help screen reader read out menu name.
- Tested main menu dropdown conditions. It will read out correctly the menu list level.

### OTHER ###
- [X] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [X] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [X] Does this pass CBT?

### STEPS TO VERIFY ###

1. Activate voiceover mode.
2. Read the entire webpage for the screen reader to read out the main menu name.
3. Click on the mobile menu button for the screen reader to read out the mobile menu name.

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

### CODE REVIEW

**Comment by Cheffheid** [comment]
I have some thoughts on this, besides the fact that the Google+ removal should probably have been done on its own branch. ;)

`<nav>` elements have an implied role of navigation already, and so adding a `role="navigation"` to it is a bit redundant. However, it's not _entirely_ weird from a compatibility perspective, but I think we've reached a point where it's not necessary anymore. 

Best I can find is that IE11 is the only browser with a broken role disclosure, but I can't actually confirm this with NVDA on my Windows 10 machine. That is to say, it still properly announces the landmark as navigation and it's included in jumping from landmark to landmark. No different from how it works in Firefox. 

And so, while it doesn't really do harm I'm not entirely sure if it's really very useful to include either. But maybe additional testing is required.

In any case, good call on the aria-labels though, should be useful!

**Comment by mharis** [comment]
@Cheffheid I think the `google+` removal in this PR is actually a merge from master.

File: header.php [hunk 1]
```diff
@@ -54,17 +54,19 @@
...
-			'fallback_cb'     => false,
-			'theme_location'  => 'primary',
-			'menu_id'         => 'primary-menu',
-			'menu_class'      => 'menu dropdown',
-			'container'       => 'nav',
-			'container_class' => 'main-navigation',
-			'container_id'    => 'site-navigation',
-		) );
-		?>
+		<nav id="site-navigation" class="main-navigation" role="navigation" aria-label="<?php esc_html_e( 'Main Menu', '_s' ); ?>">
```
Comment (coreymcollins) [inline header.php]: We can remove the `role` attribute here.

File: inc/template-tags.php [hunk 2]
```diff
@@ -525,7 +525,7 @@ function _s_display_mobile_menu() {
 	}
 	?>
 	<div class="off-canvas-screen"></div>
-	<nav class="off-canvas-container" aria-hidden="true">
+	<nav class="off-canvas-container" role="navigation" aria-label="<?php esc_html_e( 'Mobile Menu', '_s' ); ?>" aria-hidden="true">
```
Comment (coreymcollins) [inline inc/template-tags.php]: We can remove the `role` attribute here.

**Review by coreymcollins — CHANGES_REQUESTED** [review]
It looks like Jeffrey's redundancy check on the `role` attribute being used is on point: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role
>Screen readers will announce the type of role the landmark is. Because of this, you do not need to describe what the landmark is in its label. For example, a declaration of role="navigation" with an of aria-label="Primary navigation" may be announced redundantly as, "primary navigation navigation".

I think we can remove the `role` on both of these since they're using the `nav` tag.

The other thing I noticed when sizing my browser down is that the screen reading for the mobile menu isn't super great. The first issue is that it sits below the admin bar in the markup, so it's literally the very last thing we get to when reading the page which isn't super handy.

I'm wondering if there might be some additional `aria` functionality that we can use to give the menu focus when it's opened so it's read above anything else when opened? I tried playing with `aria-expanded` and `aria-modal` quickly but didn't have any luck changing the experience at all, but they may be worth exploring a bit more.

**Comment by mharis** [comment]
Hi @coreymcollins 

Thank you for the review. All suggestions are noted and I'll get back to it as soon as I get some downtime or during next #5FTF.

**Comment by coreymcollins** [comment]
Thanks for these updates!

With the screen reader recognizing the menu better, I went through and test that again. I am finding one thing that I think we can tweak.

The parent menu items get a button to expand their submenus on mobile:


The issue I'm seeing is that the button comes before the link in the markup:


What this boils down to is a weird read-order when this menu is read:


The screen reader goes:
1. Home
2. Open submenu, expanded button
3. Parent Page
4. Child Page 03
5. Child Page 04
6. Open submenu, collapsed button
7. Child Page 05
8. Page Markup And Formatting

I think it'd be best to arrange the button after the menu item text to read in a more logical order:
1. Home
2. Parent Page
3. Open submenu, expanded button
4. Child Page 03
5. Child Page 04
6. Child Page 05
7. Open submenu, collapsed button
8. Page Markup And Formatting

**Review by coreymcollins — APPROVED** [review]

**Pull request merged**